knowagedao/src/main/java/it/eng/spagobi/kpi/bo/KpiValueExecLog.java

```diff
@@ -2,7 +2,7 @@
 
 import java.util.Date;
 
-public class KpiValueExecLog implements Cloneable {
+public final class KpiValueExecLog implements Cloneable {
```

Comment:
Why declaring the entire class to be final?

Reply:
Take a look at previous answer, same situation.